corehq/apps/domain/decorators.py

```diff
@@ -443,9 +467,16 @@ def get_auth_decorator_map(
         'require_domain': require_domain,
         'allow_sessions': allow_sessions,
     }
+
```

Comment:
re:
> the only place we set `require_domain` to False is [here](https://github.com/dimagi/commcare-hq/blob/b1e0607dfb09202a28404f6ed5876d2a941da137/corehq/apps/api/resources/auth.py#L73-L73), which we can safely say does not use the `login_or_basic_or_api_key_ex` function since `allow_api_key_as_password` defaults to False.

Would it make sense to assert that here?
```suggestion

    assert not allow_api_key_as_password or require_domain, \
        "It is unexpected to allow API key as password and not require domain"
```

I assume it's not easy or very meaningful to add tests to verify that our code never does that, but if it's possible to verify with a test then that might make sense too.

Reply:
421c851b42f3372fa03d8fe033afccb5bb4f0c9f